Doc/whatsnew/3.11.rst

```diff
@@ -601,8 +601,8 @@ Build Changes
 
 * Building Python now requires a C99 ``<math.h>`` header file providing
   the following functions: ``copysign()``, ``hypot()``, ``isfinite()``,
-  ``isinf()``, ``isnan()``, ``round()``.
-  (Contributed by Victor Stinner in :issue:`45440`.)
+  ``isinf()``, ``isnan()``, ``round()``. It also requires the ``NAN`` constant.
```

Comment:
As mentioned on the issue, this is a big deal - by requiring the `NAN` constant, we'd be effectively requiring IEEE 754 floating-point, which is something that CPython has been reluctant to do to date.

I'm +1 on requiring IEEE 754 for CPython, but that should be an explicit decision that has buy-in from the rest of the core developers. So I think this needs discussion on python-dev before we can make this change.

Note that the other requirements listed here are not at all similar: those are all functions that are _required_ to be present by C99, while the ``NAN`` constant may or may not exist according to the standard.  (In the words of the standard, it's "defined if and only if the implementation supports quiet NaNs for the float type".)

Reply:
Nitpick: `isinf`, `isnan` and `isfinite` are macros, not functions.

Reply:
BTW, should this change have a separate news entry, given that the changes from issue bpo-45440 are already released?

Reply:
Gah. Sorry, this is what's new, not Misc/NEWS. Ignore me.

Reply:
It remains possible to build Python without Py_NAN: define Py_NO_NAN in pyconfig.h. I clarified that in the change documentation.